root/client/src/components/user/dashboard.js

```diff
@@ -4,28 +4,51 @@ import { PollForm } from '../poll/pollForm';
 export const Dashboard = () => {
   const [showPollForm, setShowPollForm] = useState(false);
   const [polls, setPolls] = useState([]);
+  const [editData, setEditData] = useState(null);
+  const [formMethod, setFormMethod] = useState('POST');
+
   const handleSubmit = async (e, formData) => {
-    e.preventDefault();  
-    await fetch('/polls', { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(formData) })
+    e.preventDefault();
+
+    if (formMethod === 'POST') {
+      await fetch('/polls', { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(formData) })
+    } else {
+      await fetch(`/polls/${editData.id}`, { method: 'PATCH', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(formData) })
+    }
```

Comment:
realistically, if the `formMethod` is POST it should have an isolated POST logic. I believe we can make it so that the `formMethod` can be PATCH which would then trigger separate logic so you don't need to nest everything into an if/else here

Reply:
I removed the formMethod state. Used the editData date in order to pass the correct action and method to the handle submit function in dashboard, which then will make the correct fetch request.